editor/src/components/canvas/controls/insertion-plus-button.tsx

```diff
@@ -0,0 +1,249 @@
+import { ControlProps } from './new-canvas-controls'
+import * as React from 'react'
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { colorTheme } from '../../../uuiui'
+import { useEditorState } from '../../editor/store/store-hook'
+import uuid from 'uuid'
+import { IndexPosition } from '../../../utils/utils'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import * as EP from '../../../core/shared/element-path'
+import { openFloatingInsertMenu } from '../../editor/actions/action-creators'
+
+const InsertionButtonOffset = 10
+
+interface ButtonControlProps {
+  key: string
+  positionX: number
+  positionY: number
+  lineEndX: number
+  lineEndY: number
+  isHorizontalLine: boolean
+  parentPath: ElementPath
+  indexPosition: IndexPosition
+}
+
+export const InsertionControls: React.FunctionComponent<ControlProps> = (
```

Comment:
Can you make these use `betterReactMemo` please

Reply:
Fixed in ae4e74f1.